src/dijkstra_spath.jl

```diff
@@ -96,6 +98,7 @@ end
 function set_source!{V,D}(state::DijkstraStates{V,D}, g::AbstractGraph{V}, s::V)
     i = vertex_index(s, g)
     state.parents[i] = s
+    state.hasparent[i] = true
```

Comment:
I'll change the terminating condition at L269 too; shouldn't need to maintain a (parent,child) pair?